go/service/identify.go

```diff
@@ -145,10 +147,18 @@ func (h *IdentifyHandler) identifyLiteUser(netCtx context.Context, arg keybase1.
 	return res, err
 }
 
-func (h *IdentifyHandler) Resolve3(ctx context.Context, arg string) (u keybase1.UserOrTeamLite, err error) {
-	ctx = libkb.WithLogTag(ctx, "RSLV")
-	defer h.G().CTrace(ctx, fmt.Sprintf("IdentifyHandler#Resolve3(%s)", arg), func() error { return err })()
-	return h.resolveUserOrTeam(ctx, arg)
+func (h *IdentifyHandler) Resolve3(ctx context.Context, arg keybase1.Resolve3Arg) (ret keybase1.UserOrTeamLite, err error) {
+	mctx := libkb.NewMetaContext(ctx, h.G()).WithLogTag("RSLV")
+	defer mctx.CTrace(fmt.Sprintf("IdentifyHandler#Resolve3(%+v)", arg), func() error { return err })()
+	retp := &ret
+	err = h.service.offlineRPCCache.Serve(mctx, arg.Oa, "identify.resolve3", false, arg, retp, func(mctx libkb.MetaContext) (interface{}, error) {
+		tmp, err := h.resolveUserOrTeam(mctx.Ctx(), arg.Assertion)
+		if err == nil {
+			*retp = tmp
```

Comment:
can Serve do this?

Reply:
not with type safety

Reply:
... or type reflection .....

Reply:
oh that's funny. So the offline pathway doesn't actually use this statement but yet this statement kinda ties the types all together.

Reply:
we should just switch to rust